File: src-backend/routes/contacts.js
const express = require('express');
// Importamos el cliente de Supabase usando la propiedad exportada con CommonJS
const { supabase } = require('../db/client');

const router = express.Router();

router.get('/', async (req, res) => {
  try {
    // Realizamos la consulta a la tabla 'contacts' de Supabase ordenando por 'name'
    const { data: contacts, error } = await supabase
      .from('contacts')
      .select('id, phone, name, var1, var2, source, created_at')
      .order('name', { ascending: true });

    if (error) {
      throw error;
    }

    // Retornamos la respuesta en el mismo formato que esperaba tu frontend
    res.json({ contacts });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

router.post('/', async (req, res) => {
  try {
    const { phone, name, var1, var2 } = req.body || {};

    if (!phone || !String(phone).trim()) {
      return res.status(400).json({ error: 'El campo phone es obligatorio.' });
    }

    const contactPayload = {
      phone: String(phone).trim(),
      name: name ? String(name).trim() : null,
      var1: var1 ? String(var1).trim() : null,
      var2: var2 ? String(var2).trim() : null,
      source: 'manual'
    };

    const { data: createdContact, error } = await supabase
      .from('contacts')
      .insert(contactPayload)
      .select('id, phone, name, var1, var2, source, created_at')
      .single();

    if (error) {
      throw error;
    }

    res.status(201).json({ contact: createdContact });
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

module.exports = router;

File: src-backend/routes/contacts.py
from fastapi import APIRouter, HTTPException, status
from db.client import db_client

router = APIRouter(prefix="/api/contacts", tags=["contacts"])

@router.get("")
async def get_contacts_endpoint():
    """
    Retorna el listado completo de contactos registrados en el CRM.
    Incluye campos de auditoría como el origen de importación (source) y fecha de creación.
    """
    try:
        contacts = await db_client.all(
            "SELECT id, phone, name, var1, var2, source, created_at FROM contacts ORDER BY name"
        )
        return {"contacts": contacts}
    except Exception as e:
        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
